components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -823,6 +823,10 @@ void RewardsServiceImpl::OnLedgerInitialized(
   if (!ready_->is_signaled()) {
     ready_->Signal();
   }
+
+  for (auto& observer : observers_) {
+    observer.OnRewardsInitialized(this);
```

Comment:
If `result` is not `LEDGER_OK`, should we notify observers? I'm not really sure what the right way to deal with an initialization error would be, but I suppose that since we're setting the `ready_` signal regardless of success, we should probably be consistent and call the observers too...

Looking at this from the other angle, why don't we call `callback` if initialization fails?

Reply:
Yeah, I wasn't sure about the observers question either but calling them on failure at least replicates how `OnWalletInitialized` used to work which is what this is modelled on.

Agreed that we should call `callback` no matter what, I made that change.